src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Common/PackageCollectionExtensions.cs

```diff
@@ -22,25 +22,25 @@ public static NuGetVersion[] GetPackageVersions(this IEnumerable<PackageIdentity
                 .ToArray();
         }
 
-        public static IEnumerable<IGrouping<string, NuGetVersion>> GroupById(this IEnumerable<PackageIdentity> packages)
+        public static IEnumerable<IGrouping<string, T>> GroupById<T>(this IEnumerable<T> packages) where T : PackageIdentity
         {
             return packages
-                .GroupBy(p => p.Id, p => p.Version, StringComparer.OrdinalIgnoreCase);
+                .GroupBy(p => p.Id, p => p, StringComparer.OrdinalIgnoreCase);
         }
 
-        public static PackageIdentity[] GetLatest(this IEnumerable<PackageIdentity> packages)
+        public static T[] GetLatest<T>(this IEnumerable<T> packages) where T : PackageIdentity
         {
             return packages
                 .GroupById()
-                .Select(g => new PackageIdentity(g.Key, g.MaxOrDefault()))
+                .Select(g => g.OrderByDescending(x => x.Version).FirstOrDefault())
                 .ToArray();
         }
 
-        public static PackageIdentity[] GetEarliest(this IEnumerable<PackageIdentity> packages)
+        public static T[] GetEarliest<T>(this IEnumerable<T> packages) where T : PackageIdentity
         {
             return packages
                 .GroupById()
-                .Select(g => new PackageIdentity(g.Key, g.MinOrDefault()))
+                .Select(g => g.OrderBy(x => x.Version).FirstOrDefault())
```

Comment:
The same problem exists here as with GetLatest.

Reply:
The Default will never happen, so the code is redundant, and can be written instead as:

```suggestion
                .Select(g => g.OrderBy(x => x.Version).First())
```

Reply:
Fixed. Thanks!